src/engine/bank03.asm

```diff
@@ -5684,3 +5684,229 @@ Func_f085:
 	pop de
 	ret
 ; 0xf0c3
+
+SECTION "Bank 3@725a", ROMX[$725a], BANK[$3]
+
+DebugMenuEffectViewer:
+	push af
+	push bc
+	push de
+	push hl
+	farcall SetFadePalsFrameFunc
+	xor a
+	ld [wAnimationsDisabled], a
+	farcall StartFadeToWhite
+	farcall WaitPalFading_Bank07
+	farcall Func_10d40
+	farcall SetInitialGraphicsConfiguration
+	lb de, 0, 0
+	lb bc, 20, 18
+	ld h, $00
+	ld l, $00
+	farcall FillBoxInBGMap
+	lb de, 0, 12
+	lb bc, 20, 6
+	call DrawRegularTextBoxVRAM0
+	call DebugEffectViewer_PlaceTextItems
+	call ChangeAnimationPlayerSideOnStartPress.initialize
+	call ChangeEffectNumberOnDpadPress.initialize
+	call Func_3d0d
+	push af
+	ld a, MUSIC_DUELTHEME_1
+	call SetMusic
+	pop af
+	farcall StartFadeFromWhite
+	farcall WaitPalFading_Bank07
+	ld a, $18
+	call Random
+	ld [wOppCoin], a
+.button_handling_loop
+	call DoFrame
+	call CancelAnimationOnBPress
+	call ChangeAnimationPlayerSideOnStartPress
+	call ChangeEffectNumberOnDpadPress
+	call PlayAnimationOnAPress
+	call DebugPrintAnimBufferCurPosAndSize
+	call ChangeDebugViewerStateText
+	ldh a, [hKeysPressed]
+	and PAD_SELECT
+	jr z, .button_handling_loop
+	call FinishQueuedAnimations
+	farcall StartFadeToWhite
+	farcall WaitPalFading_Bank07
+	call Func_3d16
+	farcall UnsetFadePalsFrameFunc
+	pop hl
+	pop de
+	pop bc
+	pop af
+	ret
+
+PlayAnimationOnAPress:
+	ldh a, [hKeysPressed]
+	and PAD_A
+	ret z
+	ld a, [wDebugSelectedAnimNumber]
+	and a
+	ret z
+	cp $8d ; highest number animation
+	jr z, .play
+	cp $8d
+	ret nc
+.play
+	call FinishQueuedAnimations
+	call ResetAnimationQueue
+	ld a, [wDebugSelectedAnimNumber]
+	ld [wCurAnimation], a
+	ld a, [wDebugDuelAnimationScreen]
+	ld [wDuelAnimationScreen], a
+	ld a, [wDebugAnimDuelistSide]
+	ld [wDuelAnimDuelistSide], a
+	ld a, [wDebugDuelAnimLocationParam]
+	ld [wDuelAnimLocationParam], a
+	ld a, $ff
+	call Random ; pick random damage number to show
+	ld c, a
+	ld b, $00
+	ld hl, wDuelAnimDamage
+	ld [hl], c
+	inc hl
+	ld [hl], b
+	ld a, $07
+	ld [wDuelAnimSetScreen], a
```

Comment:
I'm guessing this should be `wDuelAnimEffectiveness` (a label at the same address) because `wDuelAnimSetScreen` is only used with the values 1 and 4, but `wDuelAnimEffectiveness` has its bottom 3 bits checked by `Func_1e279`.